### DISCUSSION TITLE
How to change the textblock property in cs code?

### DISCUSSION THREAD

**Discussion by blackuGT** [AvaloniaUI/Avalonia#6065] [Q&A] [answered]
I created TextBlock control with x:Name property inside .axaml, but I can't access to this control inside code (ViewModel).

I found code like this:

`this.FindControl<control>("name");` but it's not working with "this" reference inside ViewModel.

Code like this not working as well:

`
IControl controler = ControlExtensions.FindControl<TextBlock>("statusBlock");
controler.IsVisible = true;
 `

**Comment by maxkatz6** [comment]
"this" is supposed to be a reference to the your view (control or window) when using FindControl.

**Reply by blackuGT** [reply]
Still "this" is not working inside ViewModels... I tried `ControlExtensions.FindControl<TextBlock>(IControl, string)` but I don't know which is the reference of IControl here... >(IControl, string)<

**Comment by Gillibald** [comment, marked as answer]
Your ViewModel should not know anything about your View so the MVVM way of solving this is binding a property in your ViewModel to your TextBlock.
